### ISSUE TITLE
[question] Is `consistency: :strong` supposed to rollback events dispatched in case of event handler error?

### ISSUE THREAD

**Issue by datafoo** [commanded/commanded#296]
From the [documentation](https://hexdocs.pm/commanded/events.html#consistency-guarantee) (emphasis is mine):

> Strong consistency will block the command dispatch and wait for all strongly consistent event handlers to *successfully* process all events created by the command.

In case the events handlers does not process all events *successfully*, what is supposed to happen? In my tests, I get a `{:error, :consistency_timeout}` but the event that cannot be handled is present in the event store. Is this the expected behavior? Why?

**Comment by slashdotdash** [comment]
Yes, the designed behaviour is that the events are persisted but the dispatch result will be `{:error, :consistency_timeout}`. The reason being is that the events cannot be _undone_ by a handler failing to process them as handlers are indepently running processes. 

Dispatch `consistency: :strong` is really just a tool for blocking command dispatch until handlers have processed the event(s). It doesn't affect how events are processed once persisted to the event store.

We could consider renaming the option to clarify its behaviour (e.g. `wait_for`).

**Comment by datafoo** [comment]
Ok, it make sense. Thank you.

I believe a quick explanation in the documentation could be helpful, as well as removing the word "successfully".

> Strong consistency will block the command dispatch and wait for all strongly consistent event handlers to ~~successfully~~ process all events created by the command.

Perhaps a different name for this option could also be helpful indeed.
